clinica/iotools/converters/oasis3_to_bids/oasis3_utils.py

```diff
@@ -77,28 +77,53 @@ def read_imaging_data(imaging_data_directory: PathLike) -> DataFrame:
     source_dir_series = source_path_series.apply(lambda x: Path(str(x)).parent).rename(
         "source_dir"
     )
-
     file_spec_series = source_path_series.apply(lambda x: Path(str(x)).parts[0]).rename(
         "path"
     )
-
+    source_file_series = source_path_series.apply(
+        lambda x: identify_modality(x)
```

Comment:
For consistency and respect of the type hint of the function:

```suggestion
        lambda x: identify_modality(str(x))
```